services/provider/server/server.go

```diff
@@ -201,3 +216,56 @@ func newClient() (client.Client, error) {
 
 	return client, nil
 }
+
+func (s *OCSProviderServer) getPublicKey(ctx context.Context, secretName string) (string, error) {
+	pubKeySecret := &corev1.Secret{}
+	err := s.client.Get(ctx, types.NamespacedName{Name: secretName, Namespace: s.namespace}, pubKeySecret)
+	if err != nil {
+		return "", fmt.Errorf("failed to get public key secret %q", secretName)
+	}
+
+	publicKey := string(pubKeySecret.Data["key"])
+	if publicKey == "" {
+		return "", fmt.Errorf("public key is not found inside the secret %q", secretName)
+	}
+
+	return publicKey, nil
+}
+
+func validateTicket(ticket, rawPubKey string) error {
+	block, _ := pem.Decode([]byte(rawPubKey))
+	if block == nil {
+		return fmt.Errorf("invalid PEM block")
+	}
+
+	key, err := x509.ParsePKIXPublicKey(block.Bytes)
+	if err != nil {
+		return fmt.Errorf("failed to parse public key. %v", err)
+	}
+
+	pubKey := key.(*rsa.PublicKey)
+
+	ticketArr := strings.Split(string(ticket), ".")
+	if len(ticketArr) != 0 {
+		return fmt.Errorf("invalid ticket")
+	}
+
+	message, err := base64.StdEncoding.DecodeString(ticketArr[0])
+	if err != nil {
+		return fmt.Errorf("failed to decode payload. %v", err)
+	}
+	signature, err := base64.StdEncoding.DecodeString(ticketArr[1])
+	if err != nil {
+		return fmt.Errorf("failed to decode sigature. %v", err)
+	}
+
+	hash := sha256.Sum256(message)
+	err = rsa.VerifyPKCS1v15(pubKey, crypto.SHA1, hash[:], signature)
+	if err != nil {
+		return fmt.Errorf("failed to verify onboarding ticket signature. %v", err)
+	}
+
+	klog.Info("Successfully verified onboarding ticket")
+
+	return nil
```

Comment:
Should return the unencoded payload (preferably as a go struct) so the calling function could use the validated ticket content to verify other aspects of the payload (like expiration, and caps in the future))

Reply:
using `NewOnboardingTicket` to return a go struct with decoded payload and signature.  Now `Validate` function is only concerned about the validation of the ticket.